packages/checkout/sdk/src/sdk.ts

```diff
@@ -722,4 +724,20 @@ export class Checkout {
   public async isSwapAvailable(): Promise<boolean> {
     return this.availability.checkDexAvailability();
   }
+
+  /**
+   * Fetches the approval and swap transaction details including the quote for the swap.
+   * @param {SwapParams} params - The parameters for the swap.
+   * @returns {Promise<SwapResult>} - A promise that resolves to the swap result (approvla tx, swap tx, quote).
+   */
+  public async swap(params: SwapParams): Promise<SwapResult> {
+    return swap.swap(
```

Comment:
please add this line to ensure provider was validated before starting the swap.

```
    const web3Provider = await provider.validateProvider(
      this.config,
      params.provider,
    );
    
    ...
    
    return swap.swap(this.config, web3Provider, params.fromToken, ...);
    ```

Reply:
done